R/xml_log.R

```diff
@@ -0,0 +1,102 @@
+##' @title  return datatable for parse xml data 
+##' @param log the input log file path
+##' @import xml2
+##' @export
+deepstate_xmlog <- function(log){
+  get_string <- function(node, child){
+    paste(xml2::xml_contents(xml2::xml_child(node, child)))
+  }
+  stack.trace <- function(s.node,src.dir.xpath){
+    first.dir <- xml2::xml_find_first(s.node, src.dir.xpath)
+    first.frame <- xml2::xml_parent(first.dir)
+    first.file <- get_string(first.frame, "file")
+    first.line <- get_string(first.frame, "line")
+    if(length(first.line) > 0 && length(first.file) >0){
+      return(paste0(first.file, ' : ', first.line))
+    }else{return("NA")}
+  }
+  address.trace<-function(err.node){
+    childs <- as_list(err.node)
+    names.list <- c(names(childs))
+    if(any(names.list=="auxwhat")){
+      vals <- which(names.list=="auxwhat")
+      #print("val")
+      val = (names.list[vals+1] == "stack")
+      #print(val)
+      if(!is.na(val) && val == TRUE)
+        return(1)
+    }else{return(-1)}
+  } 
```

Comment:
no need to use return at end of function.
what do the 1 and -1 signify? it is not clear.

Reply:
Not returning the value anymore. updating 1 and -1 to "stack" (if stack trace is present) else NA_character_ if no stack trace respectively.